dev_setup.sh

```diff
@@ -9,19 +9,46 @@
 # environment for cryptol. Any new environment requirements should be
 # added to this script.
 #
+# There is some half-baked support for macOS 12 (x86_64), but it hasn't been
+# tested.
+#
 
 set -e
 
 HERE=$(cd `dirname $0`; pwd)
 LOG=$HERE/dev_setup.log
 ROOT=$HERE
-ENV=$ROOT/env.sh
+
+# Create a new, empty file for environment variables
+VAR_FILE=$ROOT/env.sh
+truncate -s 0 $VAR_FILE
+echo "# Generated environment variables. Manual changes will get clobbered by dev_setup.sh" >> $VAR_FILE
 
 WHAT4_SOLVERS_SNAPSHOT="snapshot-20240212"
 WHAT4_SOLVERS_URL="https://github.com/GaloisInc/what4-solvers/releases/download/$WHAT4_SOLVERS_SNAPSHOT/"
 WHAT4_SOLVERS_MACOS_12="macos-12-X64-bin.zip"
 WHAT4_SOLVERS_MACOS_14="macos-14-ARM64-bin.zip"
 
+MACOS14="macos14"
+MACOS12="macos12"
+# Make sure we're running on a supported platform
+case $(uname -s) in
```

Comment:
I recommend putting this whole `case` into a function, `supported_platform`, whose output is the platform string. Then reduce duplication of the echo/exit pairs so that you end up with something like
```bash
CRYPTOL_PLATFORM=$(supported_platform)

if [ -z "$CRYPTOL_PLATFORM" ]
then
    echo "Unsupported platform"
    exit 1
fi
```